src/app/qgsguivectorlayertools.cpp

```diff
@@ -29,19 +29,20 @@
 #include "qgsvectordataprovider.h"
 #include "qgsvectorlayer.h"
 
-bool QgsGuiVectorLayerTools::addFeature( QgsVectorLayer *layer, const QgsAttributeMap &defaultValues, const QgsGeometry &defaultGeometry, QgsFeature *feat ) const
+bool QgsGuiVectorLayerTools::addFeature( QgsVectorLayer *layer, const QgsAttributeMap &defaultValues, const QgsGeometry &defaultGeometry, QgsFeature *feat, QWidget *parentWidget, bool showModal, bool hideParent ) const
 {
   QgsFeature *f = feat;
   if ( !feat )
     f = new QgsFeature();
 
   f->setGeometry( defaultGeometry );
-  QgsFeatureAction a( tr( "Add feature" ), *f, layer );
-  a.setForceSuppressFormPopup( forceSuppressFormPopup() );
-  const bool added = a.addFeature( defaultValues );
+  QgsFeatureAction *a = new QgsFeatureAction( tr( "Add feature" ), *f, layer, QString(), -1, parentWidget );
+  a->setForceSuppressFormPopup( forceSuppressFormPopup() );
+  const bool added = a->addFeature( defaultValues, showModal, nullptr, hideParent );
   if ( !feat )
     delete f;
-
+  if ( showModal )
```

Comment:
If user cancels the adding, the action object will remain until we close the parent (relationeditor) widget. Maybe we could delete it on addFeatureActionFinished, no?

Reply:
If it's not `showModal` the object persists at this point. But since it's parent became the dialog. It's destroyed when the dialog is destroyed. So in this use case it only needs to be deleted when `showModal` is set. But good you say it, there is a use case that could be neither modal nor set the dialog as parent. E.g. `isDisabledAttributeValuesDlg`.